packages/router/src/hooks/use_route.rs

```diff
@@ -74,10 +74,28 @@ impl UseRoute {
 /// This hook provides access to information about the current location in the
 /// context of a [`Router`]. If this function is called outside of a `Router`
 /// component it will panic.
-pub fn use_route(cx: &ScopeState) -> UseRoute {
-    let router = cx
-        .consume_context::<RouterService>()
-        .expect("Cannot call use_route outside the scope of a Router component")
-        .clone();
-    UseRoute { router }
+pub fn use_route(cx: &ScopeState) -> &UseRoute {
+    &cx.use_hook(|_| {
+        let router = cx
+            .consume_context::<RouterService>()
+            .expect("Cannot call use_route outside the scope of a Router component");
+
+        router.subscribe_onchange(cx.scope_id());
+
+        UseRouteInner {
```

Comment:
I think calling this `UseRouteInner` is confusing when it's not contained in a `UseRoute`. Maybe the existing `UseRoute` needs to be renamed to Inner and this one should be `UseRoute`?

Reply:
Hmmmm maybe something like `UseRouteLIstener` since its only job is to be dropped when the component is unmounted?

Reply:
That makes sense.